jina/hubble/helper.py

```diff
@@ -46,3 +98,63 @@ def _zip(base_path, path, archive):
     zip_stream.seek(0)
 
     return zip_stream
+
+
+def download_with_resume(
+    url: str,
+    target_dir: 'Path',
+    filename: 'Path' = None,
+    md5sum: str = None,
+) -> 'Path':
+    """
+    Download file from url to target_dir, and check md5sum.
+    Performs a HTTP(S) download that can be restarted if prematurely terminated.
+    The HTTP server must support byte ranges.
+
+    :param url: the URL to download
+    :param md5sum: the MD5 checksum to match
+    :param target_dir: the target path for the file
+    :param filename: the filename of the downloaded file
+
+    :return: the filepath of the downloaded file
+    """
+    with ImportExtensions(required=True):
+        import requests
+
+    def _download(url, target, resume_byte_pos: int = None):
+        resume_header = (
+            {'Range': f'bytes={resume_byte_pos}-'} if resume_byte_pos else None
+        )
+
+        r = requests.get(url, stream=True, headers=resume_header)
```

Comment:
Add `HubDownloadError` in `jina/excepts.py`. We need to clean up the `hub` related exceptions in that file as well.
```suggestion
        try:
            r = requests.get(url, stream=True, headers=resume_header)
        except requests.exceptions.RequestException as e:
            raise HubDownloadError(e)
```